### ISSUE TITLE
Remove 'Experimental Features' warning

### ISSUE THREAD

**Issue by mpscholten** [NixOS/nix#7000]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when nix doesn't do anything when I use "experimental features" such as nix flake or nix develop.

I always get this error:

```bash
error: experimental Nix feature 'nix-command' is disabled; use '--extra-experimental-features nix-command' to override
```

Instead I expect nix to just run it's thing. I don't care that this feature is experimental, I just want nix to run the command I told it to run. Why does everything in nix needs to be so frustrating?

**Describe the solution you'd like**

The best solution would be to remove the warning. I know what I'm doing, I don't need the program to stop me.

Another solution would be to better describe how to fix the problem. E.g. in my case I was checking out an internal repo used at my company which contained a `.envrc` file which uses the direnv nix flake integration. So in that case I cannot pass any parameters to nix, as the nix command is called by direnv itself.

**Describe alternatives you've considered**

I've tried to get rid of this error a few times in the past when trying out `nix develop`. I never got `nix develop` working (the error message is not very helpful on NixOS as well) and just gave up.

**Additional context**
OS: macOS Monterey
Nix version: `nix (Nix) 2.10.3`

**Comment by SuperSandro2000** [comment]
> Instead I expect nix to just run it's thing. I don't care that this feature is experimental, I just want nix to run the command I told it to run.

Just add ``experimental-features = flakes nix-command`` to ``nix.conf`` to always have it enabled.

> Why does everything in nix needs to be so frustrating?

Because nix is not pure magic and can do everything perfect first try but the community is working on it one step at the time.

> The best solution would be to remove the warning. I know what I'm doing, I don't need the program to stop me.

It will be, when the features are no longer experimental.

**Comment by mpscholten** [comment]
Thanks 👍 

> Just add experimental-features = flakes nix-command to nix.conf to always have it enabled.

It would be useful to have this hint in the error message itself.

> Because nix is not pure magic and can do everything perfect first try but the community is working on it one step at the time.

👍 Sorry, I was very frustrated here. I hope my intent to improve nix is clear here as well :)

> It will be, when the features are no longer experimental.

I understand these are experimental features, but what's the reason users need to opt in to use them? It feels like it just adds trouble to the workflow. Especially as these experimental nix features are now used more often.

**Comment by sellout** [comment]
I agree that the message should describe how to enable them (or at least include a link to a place that gives more context and explains how to enable them while also explaining why they're experimental.

**Comment by mpscholten** [comment]
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/2072185/192139026-1babc7f7-a363-49f4-9f32-5c7fa3142341.png">


Here's another example of this problem I just hit while building some haskell packages with `nix-shell`: Nix specifically told me to run `nix log ..`. I then tried to run `nix log ..` and this failed with the same error.

Maybe if these features are still experimental and opt-in only, nix shouldn't suggest me to run that command. Or maybe nix should tell me to run `nix log ... --extra-experimental-features nix-command` instead.

The current behaviour, something fails but nix gives me a hint, but then that hint also fails is definitely more frustrating than it needs to be :)

**Comment by SuperSandro2000** [comment]
You maybe want try to place the --extra-experimental-features between nix and the verb or right after the verb. When placing it after the flake ref it maybe doesn't work on older versions. For me on 2.24 it works with your example.

``nix --extra-experimental-features run 'flakes nix-command' nixpkgs#home-manager -- --flake gitlab:Diti/nix-dtp --help``

**Comment by dulhaver** [comment]
@SuperSandro2000
>  Just add `experimental-features = flakes nix-command` to `nix.conf` to always have it enabled.

you mean adding this to `/etc/nix/nix.conf` despite the warning ...

```
# WARNING: this file is generated from the nix.* options in
# your NixOS configuration, typically
# /etc/nixos/configuration.nix.  Do not edit it!
```
and despite the fact `/etc/nix/nix.conf` seems to be a read-only filesystem
```
[root@nixos:~]# echo "experimental-features = flakes nix-command" >> /etc/nix/nix.conf
-bash: /etc/nix/nix.conf: Read-only file system
```

I admit it hints what to do instead with `is generated from the nix.* options in [...] /etc/nixos/configuration.nix`.
However that hint is not more than a hint and it remains mysterious how and where to find those `nix.* options` part in `/etc/nixos/configuration.nix` and how this addition needs to look like practically.
